processing/src/test/java/org/apache/druid/timeline/DataSegmentTest.java

```diff
@@ -361,10 +364,48 @@ public void testWithLastCompactionState()
                                            .version(DateTimes.of("2012-01-01T11:22:33.444Z").toString())
                                            .shardSpec(getShardSpec(7))
                                            .size(0)
-                                           .build();
+                                            .build();
     Assert.assertEquals(segment1, segment2.withLastCompactionState(compactionState));
   }
 
+  @Test
+  public void testAnnotateWithLastCompactionState()
+  {
+    final CompactionState compactionState = new CompactionState(
+        new DynamicPartitionsSpec(null, null),
+        new DimensionsSpec(DimensionsSpec.getDefaultSchemas(ImmutableList.of("bar", "foo"))),
+        ImmutableList.of(ImmutableMap.of("type", "count", "name", "count")),
+        ImmutableMap.of("filter", ImmutableMap.of("type", "selector", "dimension", "dim1", "value", "foo")),
+        Collections.singletonMap("test", "map"),
+        Collections.singletonMap("test2", "map2")
+    );
+
+    final Function<Set<DataSegment>, Set<DataSegment>> annotateFn = CompactionState.compactionStateAnnotateFunction(
+        new DynamicPartitionsSpec(null, null),
+        new DimensionsSpec(DimensionsSpec.getDefaultSchemas(ImmutableList.of("bar", "foo"))),
+        ImmutableList.of(ImmutableMap.of("type", "count", "name", "count")),
+        ImmutableMap.of("filter", ImmutableMap.of("type", "selector", "dimension", "dim1", "value", "foo")),
```

Comment:
Assign these values to variables and reuse the variables in the two places for clarity that the two are actually the same.